packages/contracts-bridge/contracts/test/TokenRegistry.t.sol

```diff
@@ -1,10 +1,558 @@
+    ) public {
+        address localAddress = createRemoteToken(fuzzedDomain, fuzzedId);
+        address repr = tokenRegistry.getRepresentationAddress(
+            fuzzedDomain,
+            fuzzedId
+        );
+        assertEq(repr, localAddress);
+    }
+
+    event TokenDeployed(
+        uint32 indexed domain,
+        bytes32 indexed id,
+        address indexed representation
+    );
+
+    function test_ensureLocalTokenDeploy() public {
+        uint32 newDomain = 13;
+        bytes32 newId = bytes32("hey yoou");
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        address calculated = computeCreateAddress(address(tokenRegistry), 2);
+        vm.expectEmit(true, true, true, false);
+        emit TokenDeployed(newDomain, newId, calculated);
+        vm.prank(tokenRegistry.owner());
+        address deployed = tokenRegistry.ensureLocalToken(newDomain, newId);
+        assertEq(deployed, calculated);
+    }
+
+    function test_ensureLocalTokenExisting() public {
+        vm.prank(tokenRegistry.owner());
+        address addr = tokenRegistry.ensureLocalToken(
+            remoteDomain,
+            remoteTokenRemoteAddress
+        );
+        assertEq(addr, remoteTokenLocalAddress);
+    }
+
+    function test_ensureLocalTokenOnlyOwner() public {
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        vm.expectRevert("Ownable: caller is not the owner");
+        tokenRegistry.ensureLocalToken(remoteDomain, remoteTokenRemoteAddress);
+    }
+
+    function test_ensureLocalTokenDeployFuzzed(uint32 domain, bytes32 id)
+        public
+    {
+        vm.startPrank(tokenRegistry.owner());
+        if (domain == localDomain) {
+            assertEq(
+                tokenRegistry.ensureLocalToken(domain, id),
+                TypeCasts.bytes32ToAddress(id)
+            );
+            return;
+        }
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        address calculated = computeCreateAddress(address(tokenRegistry), 2);
+        vm.expectEmit(true, true, true, false);
+        emit TokenDeployed(domain, id, calculated);
+        address deployed = tokenRegistry.ensureLocalToken(domain, id);
+        assertEq(deployed, calculated);
+        vm.stopPrank();
+    }
+
+    function test_ensureLocalTokenOnlyOwnerFuzzed(address user) public {
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        vm.assume(user != tokenRegistry.owner());
+        vm.expectRevert("Ownable: caller is not the owner");
+        vm.prank(user);
+        tokenRegistry.ensureLocalToken(remoteDomain, remoteTokenRemoteAddress);
+    }
+
+    function test_enrollCustom() public {
```

Comment:
should have a test of enroll custom that

- mints tokens on the old repr
- enrolls a new custom token
- checks that both mappings are set correctly (r to c for old is set. r to c & c to r for new are set)
- migrates

Reply:
I am not sure this is relevant to a unit-test for `enrolLCustom` of tokenRegistry.

I think you are referring to bridgeRouter? In the unit-test I test exactly what the code does on TokenRegistry

Reply:
fair enough, that does belong in bridgerouter tests